编译ASAN版本报错

**Describe the bug (描述bug)**
尝试编译ASAN版本的brpc库，将CMakeLists.txt修改：
```
< set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -O2 -pipe -Wall -W -fPIC -fstrict-aliasing -Wno-invalid-offsetof -Wno-unused-parameter -fno-omit-frame-pointer")
---
> set(CMAKE_CXX_FLAGS "${CMAKE_CPP_FLAGS} -O1 -fsanitize=address -pipe -Wall -W -fPIC -fstrict-aliasing -Wno-invalid-offsetof -Wno-unused-parameter -fno-omit-frame-pointer")
```
链接报错：
```
[ 97%] Linking CXX executable ../../output/bin/rpc_press
`.note.GNU-stack' referenced in section `.init_array.00099' of CMakeFiles/OBJ_LIB.dir/bthread/context.cpp.o: defined in discarded section `.note.GNU-stack' of CMakeFiles/OBJ_LIB.dir/bthread/context.cpp.o
collect2: error: ld returned 1 exit status
make[2]: *** [output/lib/libbrpc.so] Error 1
make[1]: *** [src/CMakeFiles/brpc-shared.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
../../output/lib/libbrpc.a(file_path.cc.o): In function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::size() const':
/usr/include/c++/7.3.0/bits/basic_string.h:912: undefined reference to `__asan_report_load8'
```

**To Reproduce (复现方法)**
-

**Expected behavior (期望行为)**
编译链接通过

**Versions (各种版本)**
OS:
Compiler:
brpc:
protobuf:

**Additional context/screenshots (更多上下文/截图)**

> "undefined reference to `__asan_report_load8'" reports a linking error, maybe you can try to add '-lasan'.

@lorinlee Thanks! After I adding '-lasan', another link error reported: 
```
`.note.GNU-stack' referenced in section `.init_array.00099' of ../../output/lib/libbrpc.a(context.cpp.o): defined in discarded section `.note.GNU-stack' of ../../output/lib/libbrpc.a(context.cpp.o)
```

> > "undefined reference to `__asan_report_load8'" reports a linking error, maybe you can try to add '-lasan'.
> 
> @lorinlee Thanks! After I adding '-lasan', another link error reported:
> 
> ```
> `.note.GNU-stack' referenced in section `.init_array.00099' of ../../output/lib/libbrpc.a(context.cpp.o): defined in discarded section `.note.GNU-stack' of ../../output/lib/libbrpc.a(context.cpp.o)
> ```

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=47071
context.cpp里汇编写的不规范导致的，参照这个链接，在你所在平台对应的两个汇编函数结尾加上 一行".previous" "\n"就行了。

--- a/src/bthread/context.cpp
+++ b/src/bthread/context.cpp
@@ -576,6 +576,7 @@ __asm (
 ".size bthread_jump_fcontext,.-bthread_jump_fcontext\n"
 "@ Mark that we don't need executable stack.\n"
 ".section .note.GNU-stack,\"\",%progbits\n"
+".previous\n"
 );
 
 #endif
@@ -607,6 +608,7 @@ __asm (
 ".size bthread_make_fcontext,.-bthread_make_fcontext\n"
 "@ Mark that we don't need executable stack.\n"
 ".section .note.GNU-stack,\"\",%progbits\n"
+".previous\n"
 );
 
 #endif
@@ -678,6 +680,7 @@ __asm (
 ".size   bthread_jump_fcontext,.-bthread_jump_fcontext\n"
 "# Mark that we don't need executable stack.\n"
 ".section .note.GNU-stack,\"\",%progbits\n"
+".previous\n"
 );
 
 #endif
@@ -710,6 +713,7 @@ __asm (
 ".size   bthread_make_fcontext,.-bthread_make_fcontext\n"
 "# Mark that we don't need executable stack.\n"
 ".section .note.GNU-stack,\"\",%progbits\n"
+".previous\n"
 );
 
 #endif

